File: price_definition/price.py
from datetime import datetime


# holds a price definition
class ProductPriceModel:
    def __init__(self, price_date: datetime.date, price: float, price_sale: float, is_onsale: bool, is_available: bool):
        self.price_date = price_date
        self.price = price
        self.price_sale = price_sale
        self.is_onsale = is_onsale
        self.is_available = is_available

File: Spendy/SpendyBackend/Models/Price.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace SpendyBackend.Models;

[Table("prices")]
public class Price
{
    [Column("price_id")]
    public int PriceId { get; set; }
    [Column("product_id")]
    public int ProductId { get; set; }
    [Column("store_id")]
    public int StoreId { get; set; }
    [Column("price_date")]
    public DateTime PriceDate { get; set; }
    [Column("price")]
    public double PriceOriginal { get; set; }
    [Column("is_onsale")]
    public bool IsOnSale { get; set; }
    [Column("price_sale")]
    public double PriceSale { get; set; }
    [Column("is_available")]
    public bool IsAvailable { get; set; }
    
    public Product Product { get; set; }
    public Store Store { get; set; }
    
}
